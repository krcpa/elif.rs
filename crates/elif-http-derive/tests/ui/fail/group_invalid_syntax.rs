//! Test that #[group] macro fails with proper error messages on invalid syntax

use elif_http_derive::{group, get};

// This should fail with proper error message
#[group(invalid_syntax_here)]
impl BadGroup {
    #[get("/test")]
    pub fn test() -> String {
        "test".to_string()
    }
}

fn main() {}