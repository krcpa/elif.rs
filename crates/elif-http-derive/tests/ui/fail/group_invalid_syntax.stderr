error: expected string literal
 --> tests/ui/fail/group_invalid_syntax.rs:6:9
  |
6 | #[group(invalid_syntax_here)]
  |         ^^^^^^^^^^^^^^^^^^^

warning: unused import: `get`
 --> tests/ui/fail/group_invalid_syntax.rs:3:31
  |
3 | use elif_http_derive::{group, get};
  |                               ^^^
  |
  = note: `#[warn(unused_imports)]` on by default
